### DISCUSSION TITLE
Typesafe query in flutter

### DISCUSSION THREAD

**Discussion by Aadim1** [supabase/supabase#27107] [Feature Requests]
Hey, are there any plans to make the supabase_flutter library typesafe either through code genration or other methods? I believe that it will be a huge plus and improve DX. 

Has there been any discussion in the past?

**Comment by mmvergara** [comment]
Hey i made a tool to [Generate Dart classes from your Supabase schema](https://github.com/mmvergara/supadart) , maybe it can help you..


### Fetching Data Example

```dart
// fetchedBooks is a typeof List<Books>
final fetchedBooks = await supabase
      .from(Books.table_name)
      .select("*")
      .withConverter((data) => data.map(Books.fromJson).toList());
```

> Has there been any discussion in the past?
https://github.com/supabase/supabase-flutter/issues/139
